Data/NF/Unsafe.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 708
+{-# LANGUAGE PatternSynonyms #-}
+#endif
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Data.NF.Unsafe
+-- Copyright   :  (c) David Feuer 2016
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  ezyang@cs.stanford.edu
+-- Stability   :  experimental
+-- Portability :  portable
+--
+-- This module provides an unsafe pattern synonym for the type 'NF', allowing you
+-- to create values of type 'NF' without any runtime overhead if you fulfill a
+-- proof obligation that the value is already in normal form. It also provides
+-- functions allowing more general coercions relating to 'NF'.
```

Comment:
Then the names need to be disambiguated better. `mappingCoercion` is very unclear on its own.